exchanges/bitstamp/bitstamp.go

```diff
@@ -86,6 +83,40 @@ func (b *Bitstamp) GetFee(ctx context.Context, feeBuilder *exchange.FeeBuilder)
 	return fee, nil
 }
 
+// GetTradingFee returns a trading fee based on a currency
+func (b *Bitstamp) getTradingFee(ctx context.Context, feeBuilder *exchange.FeeBuilder) (float64, error) {
+	fees, err := b.GetAccountTradingFee(ctx, feeBuilder.Pair)
+	if err != nil {
+		return 0, err
+	}
+	if len(fees) != 1 {
+		return 0, errors.New("did not receive exactly one fee from api")
+	}
+	fee := fees[0].Taker
+	if feeBuilder.IsMaker {
+		fee = fees[0].Maker
+	}
+	return fee / 100 * feeBuilder.PurchasePrice * feeBuilder.Amount, nil
+}
+
+// GetAccountTradingFee returns a TradingFee for a pair
+func (b *Bitstamp) GetAccountTradingFee(ctx context.Context, pair currency.Pair) ([]MakerTakerFees, error) {
+	path := bitstampAPITradingFees + "/" + strings.ToLower(pair.String())
+
+	var resp TradingFees
+	err := b.SendAuthenticatedHTTPRequest(ctx, exchange.RestSpot, path, true, nil, &resp)
+	return resp.Fees, err
+}
+
+// GetAccountTradingFees returns a slice of TradingFee
+func (b *Bitstamp) GetAccountTradingFees(ctx context.Context) ([]TradingFees, error) {
+	path := bitstampAPITradingFees + "/"
+
+	var resp []TradingFees
+	err := b.SendAuthenticatedHTTPRequest(ctx, exchange.RestSpot, path, true, nil, &resp)
+	return resp, err
+}
```

Comment:
Can be merged into one function, also the empty pair case will need to be tested, there was missing test coverage for `GetAccountTradingFees`:
```go
// GetAccountTradingFees returns account tradings fee for a pair or all fees if pair is empty.
func (b *Bitstamp) GetAccountTradingFees(ctx context.Context, pair currency.Pair) ([]TradingFees, error) {
	path := bitstampAPITradingFees + "/" + strings.ToLower(pair.String())
	if pair.IsEmpty() {
		var resp []TradingFees
		err := b.SendAuthenticatedHTTPRequest(ctx, exchange.RestSpot, path, true, nil, &resp)
		return resp, err
	}
	var resp TradingFees
	err := b.SendAuthenticatedHTTPRequest(ctx, exchange.RestSpot, path, true, nil, &resp)
	return []TradingFees{resp}, err
}

```

Reply:
@shazbert  What I don't like about this is that `GetAccountTradingFee` is our only current use-case.

Adding `GetAccountTradingFees` is a nice-to-have, but it doesn't fill a current interface gap or use-case we have now.

So when we're making our func API harder to use for our only use-case, with the only benefit being that there's one func instead of 2, it worries me.

Now the consumer of `GetAccountTradingFees` needs to check it gets back only one, and also pull `*.Fees` out.
Arguably I'd move that check, and also the check of only 1 .Fees into `GetAccountTradingFee` to further simplify the API for consumers.

Finally, I think that having an empty pair is slightly distasteful. I'd probably use a variadic.

TL;DR: I don't think we should be afraid to have 2 funcs if it improves usability and API.
Just wanted to voice my opinion. No skin in the game.

Reply:
I tend to always default to consolidate calls with similar types and outbound request paths especially in this file, align one function to one endpoint for direct API mapping. 

I do see your point though to make it easier for users but that is why we have a common approach through the wrapper which will handle the heaving lifting of consuming this function call.

Look either way 2 functions doesn't hurt either 🤷. @gloriousCode @thrasher- Thoughts?

Reply:
The goal of {{exchange}}.go is to implement the API endpoints as close to documentation as possible. A user can use an exchange's package and know that they can call the functions that are described in the API docs. Helper functions are allowed too. Given that, an individual's desired use-case is not really a consideration for a public repository. Checking the length of a response is hardly a problem, nor is having empty params, otherwise you'll be rewriting all {{exchange}}.go implementations.

With that goal in mind, [trading fees](https://www.bitstamp.net/api/#tag/Fees/operation/GetAllTradingFees) and [trading fee for currency](https://www.bitstamp.net/api/#tag/Fees/operation/GetTradingFeesForCurrency) are two separate entries and so will be fine to have as two

Reply:
Noted, thank you @gloriousCode